### DISCUSSION TITLE
DHT22 only works with a dummy sensor

### DISCUSSION THREAD

**Discussion by GJM-76** [arendst/Tasmota#15077] [Q&A] [answered]
Hello,

I have spend all day trying to get a DHT22/AM2302 running on a nodemcu with different firmware version and i found that it will only return data if i create a dummy AM2301.
My DHT22 is connected on GPIO 5 and i created a dummy on GPIO 4 (also tested with sensor on GPIO13 with a dummy on GPIO12) and then and only then i get data from the DHT22

![dht with dummy](https://user-images.githubusercontent.com/101231979/157359375-390680ce-111f-41c7-aeca-8d1577be590f.JPG)

I googled but i could not find anything regarding this issue.
Ran tasmota versions 8.2/9.1/9.5/10 and 11 and it seems to be "in" all versions.
 
Yes i know the DHT22 is not the most reliable sensor on the market but i have them and i would really like to use them.

**Comment by ascillato** [comment, marked as answer]
Hi, 

* Have you put the 10K pull up resistor?
* Are you powering it from an stable power source?
* The power source is 3.3V or 5V?

I have been using several of these sensors with Tasmota since several years ago and have done several tests with it and some of the ways to make it fail are:
- Not having a good power source. No matter if it is 3.3V or 5V (it works fine with any of those).
- Not connecting the pull up resistor.
- Connecting the sensor to a GPIO that transmit some data at boot time like the TX pin (In this case the sensor will work until a restart of Tasmota in which the sensor is confused by the data received and will stop reading temperature).

So far the older ones (4.5 years ago) are still working fine.

Please, check the above and the wiring, and also check the GPIO limitations of the ESP8266 at https://tasmota.github.io/docs/Expanding-Tasmota/#gpio-overview and https://tasmota.github.io/docs/AM2301/

Also try to use your AM2302 defining the GPIO as `SI7021`. Please, use latest Tasmota version.

**Reply by GJM-76** [reply]
The SI7021 was the golden tip.. thank you
